.github/workflows/ci.yml

```diff
@@ -17,22 +17,17 @@ jobs:
       - name: Checkout OperationsGateway API
         uses: actions/checkout@v3
 
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+
       # Install dependencies of python-ldap
       - name: Install python-ldap dependencies
         run: |
           sudo apt-get update
           sudo apt-get install -y libsasl2-dev libldap2-dev libssl-dev
 
-      # Setup Python and environment dependencies (via cache)
-      - name: Setup Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.11
-      - name: Load Pip cache
-        uses: actions/cache@v3
-        with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-3.11-${{ env.pythonLocation }}-${{ hashFiles('.github/ci_requirements.txt') }}
```

Comment:
As a consequence of removing this, I don't think we're caching the wheels for `poetry` or `nox`. E.G. in the CI on main: https://github.com/ral-facilities/operationsgateway-api/actions/runs/13674020508/job/38230395518#step:5:13
we used cached versions, but in this CI we do a download: https://github.com/ral-facilities/operationsgateway-api/actions/runs/13708564861/job/38339723716#step:4:13

In principle, this should take longer, but actually once you factor in the time to load from cache it only looks like 1 second difference from that example. So I don't think it's a big deal either way, just wanted to check it was intentional to remove this caching.